### DISCUSSION TITLE
Simplest Map App -> DatabaseRenderer: NullPointerException

### DISCUSSION THREAD

**Discussion by Zimbelstern** [mapsforge/mapsforge#1225] [Q&A] [answered]
Hi! I have been spending the last days trying to get a MWE of an Android app using mapsforge. What I've tried so far:
- Creating a new project in Android studio
- Adding the following lines to the dependencies in `build.gradle`:

```    implementation 'org.mapsforge:mapsforge-core:0.16.0'
    implementation 'org.mapsforge:mapsforge-map:0.16.0'
    implementation 'org.mapsforge:mapsforge-map-reader:0.16.0'
    implementation 'org.mapsforge:mapsforge-themes:0.16.0'
    implementation 'net.sf.kxml:kxml2:2.3.0'
    implementation 'org.mapsforge:mapsforge-map-android:0.16.0'
    implementation 'com.caverock:androidsvg:1.4'
```
- Copying the _SimplestMapViewer_ activity to my project, adjusting the package name, copying _mapviewer.xml_ to my project
- Uploading _berlin.map_ to `/sdcard/Android/com.example.project/files/berlin.map`

The activity shows a white screen, a tiny scale bar, attribution and "working" zoom controls, but no map. The console shows a lot of these messages:

`W/DatabaseRenderer: Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.FileInputStream android.content.Context.openFileInput(java.lang.String)' on a null object reference`

Could anyone give me some hints why this is not working and show me the right way? Perhaps I am missing some basic concepts. Thanks.

**Comment by devemux86** [comment, marked as answer]
See the [getting started example](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) for how to use the library.

If you want to use a regular file path, then the storage location must be readable by your app.
Something difficult with Android scoped storage.
